x/slashing/tick.go

```diff
@@ -16,29 +16,29 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, sk Keeper) (tags
 	binary.LittleEndian.PutUint64(heightBytes, uint64(req.Header.Height))
 	tags = sdk.NewTags("height", heightBytes)
 
-	// Deal with any equivocation evidence
+	// Iterate over all the validators  which *should* have signed this block
+	for _, signingValidator := range req.Validators {
+		present := signingValidator.SignedLastBlock
+		pubkey, err := tmtypes.PB2TM.PubKey(signingValidator.Validator.PubKey)
```

Comment:
Can there be a comment here regarding what the goal of this loop is? Is it just to figure out which validators signed the block? (wasn't clear to me) If so, is this for liveness slashing?

Reply:
Yes; for downtime slashing, added a comment.